module/documents/actor.mjs

```diff
@@ -118,6 +118,86 @@ export default class ActorEd extends Actor {
     this.#processRoll( roll );
   }
 
+  /**
+   * @description                 Roll an Ability. use {@link RollPrompt} for further input data.
+   * @param {object} ability      ability.
+   * @param {object} options      Any additional options for the {@link EdRoll}.
+   */
+  async rollAbility( ability, options = {} ) {
+    const attributeStep = this.system.attributes[ability.system.attribute].step;
+    const abilityFinalStep = attributeStep + ability.system.level;
+    const difficulty = await this.getTarget( ability );
+    const edRollOptions = new EdRollOptions( {
+      rollType: "action",
+      step: { base: abilityFinalStep },
+      strain: { base: ability.system.strain},
+      target: { base: difficulty },
+      karma: { pointsUsed: this.system.karma.useAlways ? 1 : 0, available: this.system.karma.value, step: this.system.karma.step },
+      devotion: { pointsUsed: ability.system.devotionRequired ? 1: 0, pointsRequired: ability.system.devotionRequired, available: this.system.devotion.value, step: this.system.devotion.step },
+      chatFlavor: ability.name + " Test",
+    } );
+    const roll = await RollPrompt.waitPrompt( edRollOptions, options );
+    this.#processRoll( roll );
+  }
+
+  async getTarget( ability ) {
```

Comment:
I don't think that this should be part of the Actor Document. Since all  the logic depends on the system data of the ability, it should go there.

Generally as a rule of thumb: If you don't need to use `this` in the function, it probably doesn't need to be a class method. (Except static methods which have some other connection to the class)